src/System.Runtime/tests/System/ValueTypeTests.cs

```diff
@@ -16,10 +16,372 @@ public static void ToStringTest()
+            obj1.value2.value2 = float.NaN;
+
+            StructWithFloatFieldNestedTightlyPacked obj2 = new StructWithFloatFieldNestedTightlyPacked();
+            obj2.value1.value1 = 1;
+            obj2.value2.value2 = -float.NaN;
+
+            Assert.Equal(true, obj1.Equals(obj2));
+        }
+
+        [Fact]
+        public static void StructWithoutNestedOverriddenEqualsCompareTest()
+        {
+            StructWithoutNestedOverriddenEqualsAndGetHashCode obj1 = new StructWithoutNestedOverriddenEqualsAndGetHashCode();
+            obj1.value1.value = 1;
+            obj1.value2.value = 2;
+
+            StructWithoutNestedOverriddenEqualsAndGetHashCode obj2 = new StructWithoutNestedOverriddenEqualsAndGetHashCode();
+            obj2.value1.value = 1;
+            obj2.value2.value = 2;
+
+            Assert.Equal(true, obj1.Equals(obj2));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The fix was made in coreclr that is not in netfx. See https://github.com/dotnet/coreclr/issues/11452")]
+        public static void StructWithNestedOverriddenEqualsCompareTest()
+        {
+            StructWithNestedOverriddenEqualsAndGetHashCode obj1 = new StructWithNestedOverriddenEqualsAndGetHashCode();
+            obj1.value1.value = 1;
+            obj1.value2.value = 2;
+
+            StructWithNestedOverriddenEqualsAndGetHashCode obj2 = new StructWithNestedOverriddenEqualsAndGetHashCode();
+            obj2.value1.value = 1;
+            obj2.value2.value = 2;
+
+            Assert.Equal(false, obj1.Equals(obj2));
+        }
+
+        [Fact]
+        public static void StructContainsPointerCompareTest()
+        {
+            StructContainsPointer obj1 = new StructContainsPointer();
+            obj1.value1 = 1;
+            obj1.value2 = 0.0;
+
+            StructContainsPointer obj2 = new StructContainsPointer();
+            obj2.value1 = 1;
+            obj2.value2 = -0.0;
+
+            Assert.Equal(true, obj1.Equals(obj2));
+        }
+
         public struct S
         {
             public int x;
             public int y;
         }
+
+        public struct StructWithDoubleFieldNotTightlyPacked
+        {
+            public int value1;
+            public double value2;
+        }
+
+        public struct StructWithDoubleFieldTightlyPacked
+        {
+            public double value1;
+            public double value2;
+        }
+
+        public struct StructWithDoubleFieldNestedNotTightlyPacked
+        {
+            public StructWithDoubleFieldNotTightlyPacked value1;
+            public StructWithDoubleFieldNotTightlyPacked value2;
+        }
+
+        public struct StructWithDoubleFieldNestedTightlyPacked
+        {
+            public StructWithDoubleFieldTightlyPacked value1;
+            public StructWithDoubleFieldTightlyPacked value2;
+        }
+
+        public struct StructWithFloatFieldNotTightlyPacked
+        {
+            public long value1;
```

Comment:
Following the pattern of `StructWithDoubleFieldNotTightlyPacked`, I would have expected to put float before long.

StructWithFloatFieldNotTightlyPacked 
--------
----

StructWithDoubleFieldNotTightlyPacked
----
--------

- = 1 byte